scalableminds/webknossos
# Allow Exploring Datasets on Local File System

- Unifies FileSystemDataVault with other DataVaults, removes FileSystemVaultPath. VaultPaths now use  `file://` uris with absolute paths for the local file system.
- Allows exploring datasets in the local file system, in directories listed in the new config key `datastore.localFolderWhitelist`
- Note that this introduces `file://` uris also in the resulting datasource-properties.jsons. Those will not be sensibly explorable by the `WebknossosZarrExplorer` but I guess that isn’t a use case anyway

### Steps to test:
- Add a directory to config setting `datastore.localFolderWhitelist`
- Put dataset directory in there (or into its children/descendents)
- Explore that dataset in the Add Datasets view, prepend `file://` to the absolute path (so there are then three consecutive slashes)
- Should work
- Should not work is the directory is not in the whitelist
- Other datasets with mag locators (everything that isn’t wkw) should still work (with relative path + with remote path)

### TODOs:
- [x] Unify FileSystemDataVault with other DataVaults, remove FileSystemVaultPath
- [x] Allow exploring locally
- [x] Add config option to whitelist directories in which webknossos may read
  - [x] Use in datastore when reading from data vault
  - [x] Use in webknossos when exploring
- [x] Test that this does not break existing datasource-properties.jsons
  - [x] with local relative path
  - [x] with remote path

### Issues:
- fixes #7387

------
